CMakeLists.txt

```diff
@@ -1,15 +1,9 @@
 cmake_minimum_required(VERSION 3.16)
 
-# Not ideal to use this global variable, but necessary to make sure
-# that tooling and projects use the same version
+# set a default CXX standard for the tools and targets that do not specify them. 
+# If commented, the latest supported standard for your compiler is automatically set.
```

Comment:
If the next line gets commented out, the build will break, because this setting is forwarded to the `project_options` interface target below. This was the only way I could find to keep tools/targets in sync without giving the user unexpected errors..

Reply:
You don't need to manually set standards after this for each target. That is only needed if things are going to be different (which is discouraged due to ABI/API issues.)

project_options defaults to the latest standard automatically if this is missing
https://github.com/cpp-best-practices/project_options/blob/main/src/Standards.cmake